### DISCUSSION TITLE
I would like to pass cucumber configurations from RuntimeOptions. I passed runtTimeOptions as below in my CustomCucumberListner.java but those configurations are not picking up by cucumber. Should we as pass runtimeOptions to another class as an argument?. Please hele me out

### DISCUSSION THREAD

**Discussion by SaiKumarSappa** [cucumber/common#2214] [Q&A Java]
public class CucumberCustomListner implements EventListener {

	private RuntimeOptions runtimeOptions = null;

	public CucumberCustomListner() {
		runtimeOptions = new RuntimeOptionsBuilder().setMonochrome(true).addPluginName("pretty")
				.addPluginName("json:target/cucumber/cucumber.json")
				.addPluginName("html:target/cucumber/cucumber-html-report.html")
				.addPluginName("com.epam.reportportal.cucumber.ScenarioReporter").build();
	}

	@Override
	public void setEventPublisher(EventPublisher publisher) {
		publisher.registerHandlerFor(TestRunFinished.class, event -> {
			try {
				onTestRunFinished(event);
			} catch (FrameworkException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
		});
	}

	private void onTestRunFinished(TestRunFinished event) throws FrameworkException {
		//code
	}
}

**Comment by mpkorstanje** [comment]
What is the end result you are trying to achieve? Are you looking to configure a plugin? Or are you looking to configure your Cucumber execution programmatically?

**Reply by SaiKumarSappa** [reply]
Yes, I would like to configure cucumber execution programmatically. I do have **report portal** and other custom plugins are there in my framework. I would like to control this plugins loading programmatically .

**Reply by mpkorstanje** [reply]
Cucumbers support for programmatic execution is rather limited. But if you are using the `cucumber-junit-platform-engine` you can programmatically set all the properties. JUnit 5 will then pass these to Cucumber.

It also provides a good API to discover and select your tests.

* https://github.com/cucumber/cucumber-jvm/tree/main/cucumber-junit-platform-engine#using-the-junit-platform-launcher-api
* https://junit.org/junit5/docs/current/user-guide/#launcher-api

**Reply by SaiKumarSappa** [reply]
Is there any way we can achieve programmatic execution with RuntimeOptions and RuntimeOptionsBuilder? and any documentions around it pls.

I have gone through the source code and gatherd little info but I feel I need experts support here.

**Reply by mpkorstanje** [reply]
Those exists mostly to facilitate Cucumbers internal needs and they're not designed for external consumption.  So your experience will be quite rough. But if you must, the implementation of the CLI's [Main](https://github.com/cucumber/cucumber-jvm/blob/main/cucumber-core/src/main/java/io/cucumber/core/cli/Main.java) would be a good example to reference.
